src/Microsoft.AspNet.Mvc.DataAnnotations/DataAnnotationsModelValidator.cs

```diff
@@ -22,13 +30,39 @@ public DataAnnotationsModelValidator(ValidationAttribute attribute, IStringLocal
 
             Attribute = attribute;
             _stringLocalizer = stringLocalizer;
+            _validationAttributeAdapterProvider = validationAttributeAdapterProvider;
         }
 
+        /// <summary>
+        /// The attribute being validated against.
+        /// </summary>
         public ValidationAttribute Attribute { get; }
 
+        /// <summary>
+        /// Validates the context against the <see cref="ValidationAttribute"/>.
+        /// </summary>
+        /// <param name="validationContext">The context being validated.</param>
+        /// <returns>An enumerable of the validation results.</returns>
         public IEnumerable<ModelValidationResult> Validate(ModelValidationContext validationContext)
         {
-            var metadata = validationContext.Metadata;
+            if (validationContext == null)
+            {
+                throw new ArgumentNullException(nameof(validationContext));
+            }
+            if (validationContext.ModelMetadata == null)
+            {
+                throw new ArgumentException(Resources.FormatPropertyOfTypeCannotBeNull(
+                    nameof(validationContext.ModelMetadata),
+                    nameof(validationContext)));
+            }
+            if (validationContext.MetadataProvider == null)
+            {
+                throw new ArgumentException(Resources.FormatPropertyOfTypeCannotBeNull(
+                    nameof(validationContext.MetadataProvider),
+                    nameof(validationContext)));
```

Comment:
Use the message correctly and pass the second parameter into the `ArgumentException` constructor i.e.
``` c#
    typeof(ModelValidationContext)),
    nameof(validationContext));
```
Same a few lines above. (Note the above uses the type's full name. We aren't consistent about that unfortunately.)